vendor/github.com/ethereum/go-ethereum/whisper/whisperv5/api.go

```diff
@@ -240,11 +241,47 @@ func (api *PublicWhisperAPI) Post(ctx context.Context, req NewMessage) (bool, er
 	var (
 		symKeyGiven = len(req.SymKeyID) > 0
 		pubKeyGiven = len(req.PublicKey) > 0
+		isP2P       = len(req.TargetPeer) > 0
 		err         error
 	)
 
+	if api.w.deliveryServer != nil {
+		switch isP2P {
+		case true:
+			api.w.deliveryServer.SendP2PState(P2PMessageState{
+				Source:    req,
+				Status:    message.PendingStatus,
+				Direction: message.OutgoingMessage,
+			})
+		case false:
+			api.w.deliveryServer.SendRPCState(RPCMessageState{
+				Source:    req,
+				Status:    message.PendingStatus,
+				Direction: message.OutgoingMessage,
+			})
+		}
+	}
+
 	// user must specify either a symmetric or an asymmetric key
 	if (symKeyGiven && pubKeyGiven) || (!symKeyGiven && !pubKeyGiven) {
+		if api.w.deliveryServer != nil {
+			switch isP2P {
```

Comment:
If it's a boolean then `if` is ok.

Reply:
I honestly would prefer switch than `if{}...else{}`.

Reply:
@influx6 why so, btw? It feels unnatural on the first sight, `if isP2P {} else {}` seems to me to be more common and simple.